### PR TITLE
Add functionality to recipe repository

### PR DESCRIPTION
**Pull request by cagallo**
#### What's this PR do?
-  Add iterators to find recipe by name or ingredients

#### Where should the reviewer start?

- RecipeRepository.js

#### How should this be manually tested?

- N/A
- 
#### Any background context you want to provide?

- N/A

#### What are the relevant tickets?

- N/A

#### Screenshots (if appropriate)

- N/A

#### Questions:
How do we finish this?

### CODE REVIEW

**Review by kammon10 — COMMENTED** [review]
Great progress!

**Review by aliroemhildt — APPROVED** [review]
almost there!

File: test/RecipeRepository-test.js [hunk 1]
```diff
@@ -1,344 +1,346 @@
...
-          }
-        ],
-        "name": "Dirty Steve's Original Wing Sauce",
-        "tags": [
-          "sauce"
-        ]
-      }
-    ]
-    repository = new RecipeRepository(sampleData, ingredientsData);
+    // sampleData = [
```
Comment (anewb87) [inline test/RecipeRepository-test.js:13]: This sampling of data wasn't working for us while building out our tests- not big enough. Currently we have imported ALL the data, but as of right now, the plan is to create a new file for a larger sampling and import that in for testing.

File: src/classes/RecipeRepository.js [hunk 2]
```diff
@@ -40,8 +40,40 @@ class RecipeRepository {
     };
     addRecipes();
 
+    let matchingNames = this.recipeData.filter((recipe) => {
```
Comment (anewb87) [inline src/classes/RecipeRepository.js:43]: Making good headway on this next method. Feels like we're almost there.

**Review by anewb87 — APPROVED** [review]

**Review by kammon10 — APPROVED** [review]

File: src/classes/RecipeRepository.js (hunk 2 above)
Comment (cagallo) [reply src/classes/RecipeRepository.js:43]: We got this! Time to go onto Recipe!

**Pull request merged**